src/core/lib/event_engine/iomgr_engine/closure.h

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_CORE_LIB_EVENT_ENGINE_IOMGR_ENGINE_CLOSURE_H
+#define GRPC_CORE_LIB_EVENT_ENGINE_IOMGR_ENGINE_CLOSURE_H
+#include <grpc/support/port_platform.h>
+
+#include <functional>
+#include <utility>
+
+#include "absl/status/status.h"
+#include "absl/utility/utility.h"
+
+#include <grpc/event_engine/event_engine.h>
+
+namespace grpc_event_engine {
+namespace iomgr_engine {
+
+class IomgrEngineClosure final
```

Comment:
The callbacks for notify on read, notify on write take an absl::Status as argument - this is important for the tcp code to function correctly. We need a custom closure type because the default EventEngine::Closure type doesn't provide a way to pass a status when the callback is run.

Reply:
Let's add a comment describing why this exists